File: backend/Concord/Concord.Application/Models/StarWalletResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class StarWalletResponse
{
    [JsonPropertyName("Balance")]
    public int Balance { get; set; }

    [JsonPropertyName("PremiumTrialActive")]
    public bool PremiumTrialActive { get; set; }

    [JsonPropertyName("PremiumTrialExpiresAt")]
    public DateTime? PremiumTrialExpiresAt { get; set; }

    [JsonPropertyName("HasActivePremium")]
    public bool HasActivePremium { get; set; }
}

File: mobile/concord/lib/api/models/star_wallet_response.dart
import 'json_utils.dart';

class StarWalletResponse {
  const StarWalletResponse({
    required this.balance,
    required this.premiumTrialActive,
    required this.premiumTrialExpiresAt,
    required this.hasActivePremium,
  });

  factory StarWalletResponse.fromJson(Map<String, dynamic> json) {
    return StarWalletResponse(
      balance: json.field('Balance') as int? ?? 0,
      premiumTrialActive: json.field('PremiumTrialActive') as bool? ?? false,
      premiumTrialExpiresAt: parseNullableDateTime(json.field('PremiumTrialExpiresAt')),
      hasActivePremium: json.field('HasActivePremium') as bool? ?? false,
    );
  }

  final int balance;
  final bool premiumTrialActive;
  final DateTime? premiumTrialExpiresAt;
  final bool hasActivePremium;

  StarWalletResponse copyWith({int? balance, bool? premiumTrialActive, DateTime? premiumTrialExpiresAt, bool? hasActivePremium}) {
    return StarWalletResponse(
      balance: balance ?? this.balance,
      premiumTrialActive: premiumTrialActive ?? this.premiumTrialActive,
      premiumTrialExpiresAt: premiumTrialExpiresAt ?? this.premiumTrialExpiresAt,
      hasActivePremium: hasActivePremium ?? this.hasActivePremium,
    );
  }
}
